internal/cli/runner_uninstall.go

```diff
@@ -149,17 +147,30 @@ func (c *RunnerUninstallCommand) Run(args []string) int {
 	s = sg.Add("Forgetting runner %q on server...", c.id)
 	defer func() { s.Abort() }()
 
-	_, err = c.project.Client().ForgetRunner(ctx, &pb.ForgetRunnerRequest{
-		RunnerId: c.id,
-	})
+	runners, err := c.project.Client().ListRunners(ctx, &pb.ListRunnersRequest{})
 	if err != nil {
-		s.Update("Couldn't forget runner: %s", clierrors.Humanize(err))
-		s.Status(terminal.StatusWarn)
-	} else {
-		s.Update("Runner %q forgotten on server", c.id)
-		s.Status(terminal.StatusOK)
+		c.ui.Output("Error while listing runners: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// We loop and check for the runner in the list of runners, so we don't
+	// fail if it doesn't exist. We allow this so that users can "uninstall"
+	// runners which failed to be installed, and never registered with the server
+	for _, runner := range runners.Runners {
+		if runner.Id == c.id {
+			_, err = c.project.Client().ForgetRunner(ctx, &pb.ForgetRunnerRequest{
+				RunnerId: c.id,
+			})
+			if err != nil {
+				s.Update("Couldn't forget runner: %s", clierrors.Humanize(err))
+				s.Status(terminal.StatusWarn)
```

Comment:
This should probably be a ui.Output. Otherwise the next iteration, for either case, it'll get wiped due to how StepGroups work and the err will be lost.

Reply:
@briancain wouldn't that next iteration only happen if 2 runners have the same ID?

Reply:
Ah true, should we `break` out of this loop them for the `if runner.Id == c.id` case?

Reply:
+1 to return 1, and I added a `break` at the conclusion of the `if` statement, since we don't need to continue looping if we found and successfully forgot the runner.